bd-logger/src/async_log_buffer.rs

```diff
@@ -245,6 +312,15 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
         .max_size(),
     );
 
+    // Larger channel for state updates as they are less frequent and we want
+    // to avoid dropping any state updates if possible.
+    // Note that the 10 MB is not pre-allocated memory, just the upper limit of data stored within
+    // the buffer before backpressure is applied.
+    let (state_tx, state_rx) = channel(
+      1000,
+      10 * 1024 * 1024, // 10 MB
+    );
```

Comment:
I question how useful it is to pre-allocate the lots when we are tracking size. Seems like maybe we should move to unbounded channel and just track the size only? Seems easier to reason about in general both here and for logs. The only reason I can think of to keep the slot pre-allocation is that when we get around to fixing the signal handler to flush all this the unbounded will probably try to free stuff and that won't go well unless we do something fully custom, but we could cross the bridge at that point.